app/controllers/user_controller.py

```diff
@@ -39,3 +39,33 @@ def get_following(_, username: str) -> Any:
 
     following = [following.serialize_as_follower() for following in user.get_following()]
     return jsonify(following=following)
+
+
+@user_routes.route('/follow', methods=['GET', 'POST'])
+@login_required
+def follow_user_api(user_id: int) -> Any:
+    try:
+        user_id_to_follow = int(request.args.get('id'))
+    except ValueError:
+        return '', 404
+    user = User.get_by_user_id(user_id)
+    user_to_follow = User.get_by_user_id(user_id_to_follow)
+    if user and user_to_follow and user_id != user_id_to_follow:
+        User.add_to_following(user_id, user_id_to_follow)
+        return jsonify(user_id=user_id, user_id_to_follow=user_id_to_follow, following=True)
+    return '', 404
```

Comment:
We can optimize this api by removing all the user queries and instead just letting the db throw an error and catching it if there is any mismatch in the values.
Also we want to send this data through the body not query parameters since its more sensitive user_id data, and limit it to just post requests.

```suggestion
@user_routes.route('/follow', methods=['POST'])
@login_required
def follow_user(user_id: int) -> Any:
    user_id_to_follow: int = request.form.values()

    if user_id == user_id_to_follow:
        return '', 400

    try:
        User.add_to_following(user_id, user_id_to_follow)
        return '', 204
    except Exception:
        return '', 404
```

Reply:
I'll modify the code since we don't care about verification before calling the db function.